modules/SRanipal_module/src/SRanipalModule.cpp

```diff
@@ -44,24 +45,19 @@ bool SRanipalModule::configure(yarp::os::ResourceFinder &rf)
         }
     }
 
-    if (m_useEyebrows || m_useEyelids || m_useGaze)
+    if (m_useGaze || m_useAdvancedJoypad)
     {
+        m_VRInterface = std::make_shared<VRInterface>();
 
-        if (m_useGaze)
-        {
-            if (!m_gazeRetargeting.configure(rf))
-            {
-                yError() << "[SRanipalModule::configure] Failed to configure the gaze retargeting.";
-                return false;
-            }
-
-            defaultPeriod = 0.01; // Since we use velocity control for the gaze, we use faster loops
-            yInfo() << "[SRanipalModule::configure] Controlling the gaze.";
-        } else
+        if (!m_VRInterface->configure(rf))
         {
-            yInfo() << "[SRanipalModule::configure] Skipping gaze control.";
+            yError() << "[SRanipalModule::configure] Failed to configure VR interface.";
+            return false;
         }
+    }
 
+    if (m_useEyebrows || m_useEyelids || m_useGaze || m_useAdvancedJoypad)
```

Comment:
I would avoid nesting the `if`  (this should improve the readability) 
```cpp
    if (!m_useEyebrows && !m_useEyelids && !m_useGaze && !m_useAdvancedJoypad)
```

Reply:
Not sure I understood. The version you are proposing has the opposite result. Right now I want to get into that if if any of those components is active